XenAdmin/TabPages/SnapshotsPage.cs

```diff
@@ -426,16 +426,7 @@ private IList<VM> RefreshDataGridView(IList<VM> snapshots)
             for (int i = 0; i < snapshots.Count; i++)
             {
                 VM snapshot = snapshots[i];
-                if (!(snapshot.is_snapshot_from_vmpp && !toolStripMenuItemScheduledSnapshots.Checked))
-                {
-                    snapshot.PropertyChanged -= snapshot_PropertyChanged;
-                    snapshot.PropertyChanged += snapshot_PropertyChanged;
-                    //Build DataGridView
-                    SnapshotDataGridViewRow row = new SnapshotDataGridViewRow(snapshot);
-                    row.Tag = snapshot;
-                    DataGridView.Rows.Add(row);
-                }
-                else if (!(snapshot.is_vmss_snapshot && !toolStripMenuItemScheduledSnapshots.Checked))
+                if (!((snapshot.is_snapshot_from_vmpp || snapshot.is_vmss_snapshot) && !toolStripMenuItemScheduledSnapshots.Checked))

```

Comment:
there are 4 cases 

scheduled_snapshots - checkbox_value - display (outcome) 
0                                  0                             1
0                                  1                              0
1                                  0                             1
1                                   1                           1

if you take the second case where the snapshot is not a scheduled one, and the checkbox is true (which means display only scheduled snapshots) the suggested expression will evaluate to true and display the snapshot which is not desired. can you please share your thoughts on this ?